testkit/scaladsl/src/main/scala/com/lightbend/lagom/scaladsl/testkit/ServiceTest.scala

```diff
@@ -152,14 +152,17 @@ object ServiceTest {
       result match {
         case asyncResult: Future[_] =>
           import testServer.executionContext
-          asyncResult.onComplete { _ =>
-            testServer.stop()
-          }
+          // whether the future `asyncResult` was successful or failed, stop the server.
+          asyncResult.andThen {
+            case theResult => {
+              testServer.stop()
+              theResult
+            }
+          }.asInstanceOf[R]
```

Comment:
I would remove the extra curly-braces on the `case theResult` block. It's not needed and only adds noisy.
(IntelliJ gives a warning for that)